plugins/provisioners/chef/provisioner/chef_solo.rb

```diff
@@ -87,7 +87,7 @@ def expanded_folders(paths, appended_folder=nil)
                 key         = Digest::MD5.hexdigest(local_path)
                 remote_path = "#{guest_provisioning_path}/#{key}"
               else
-                @machine.ui.warn(I18n.t("vagrant.provisioners.chef.cookbook_folder_not_found_warning",
+                @machine.ui.warn(I18n.t("vagrant.provisioners.chef.#{appended_folder}_folder_not_found_warning",
```

Comment:
This is great! I think I'd like one last change. In the case that `appended_folder` was not provided and is nil, we should default to `cookbook_folder` so that users don't see an exception from the i18n library. So basically

```ruby
else
  appended_folder = "cookbook_folder" if appended_folder.nil?
  @machine.ui.warn(I18n.t("vagrant.provisioners.chef.#{appended_folder}_folder_not_found_warning",
    path: local_path.to_s))
end
```